Directory.Build.props

```diff
@@ -17,4 +17,75 @@
+    <ApprovalTestsTargetFrameworks>net461</ApprovalTestsTargetFrameworks>
+  </PropertyGroup>
+
+  <!-- .NET Core 3 / .NET Standard 2.0 -->
+  <PropertyGroup Condition=" '$(TargetFrameworkSet)' == 'netcore3.0' ">
+    <FakeItEasyTargetFrameworks>netstandard2.1</FakeItEasyTargetFrameworks>
+    <ValueTaskExtensionsTargetFrameworks>netstandard2.1</ValueTaskExtensionsTargetFrameworks>
+    <ExampleTargetFrameworks>netcoreapp3.0</ExampleTargetFrameworks>
+    <SpecsTargetFrameworks>netcoreapp3.0</SpecsTargetFrameworks>
+    <IntegrationTestsTargetFrameworks>netcoreapp3.0</IntegrationTestsTargetFrameworks>
+    <UnitTestsTargetFrameworks>netcoreapp3.0</UnitTestsTargetFrameworks>
+    <AnalyzerTestsTargetFrameworks>netcoreapp3.0</AnalyzerTestsTargetFrameworks>
+    <TestHelpersTargetFrameworks>netstandard2.0</TestHelpersTargetFrameworks>
+    <ApprovalTestsTargetFrameworks>netcoreapp3.0</ApprovalTestsTargetFrameworks>
+  </PropertyGroup>
+
+  <!-- .NET Core 2.1 / .NET Standard 2.0 -->
+  <PropertyGroup Condition=" '$(TargetFrameworkSet)' == 'netcore2.1' ">
+    <FakeItEasyTargetFrameworks>netstandard2.0</FakeItEasyTargetFrameworks>
+    <ValueTaskExtensionsTargetFrameworks>netstandard2.0</ValueTaskExtensionsTargetFrameworks>
+    <ExampleTargetFrameworks>netcoreapp2.1</ExampleTargetFrameworks>
+    <SpecsTargetFrameworks>netcoreapp2.1</SpecsTargetFrameworks>
+    <IntegrationTestsTargetFrameworks>netcoreapp2.1</IntegrationTestsTargetFrameworks>
+    <UnitTestsTargetFrameworks>netcoreapp2.1</UnitTestsTargetFrameworks>
+    <AnalyzerTestsTargetFrameworks>netcoreapp2.1</AnalyzerTestsTargetFrameworks>
+    <TestHelpersTargetFrameworks>netstandard2.0</TestHelpersTargetFrameworks>
+    <ApprovalTestsTargetFrameworks>netcoreapp2.1</ApprovalTestsTargetFrameworks>
+  </PropertyGroup>
+
+  <!-- All .NET Core / .NET Standard TFMs -->
+  <PropertyGroup Condition=" '$(TargetFrameworkSet)' == 'netcoreall' ">
+    <FakeItEasyTargetFrameworks>netstandard1.6;netstandard2.0;netstandard2.1</FakeItEasyTargetFrameworks>
+    <ValueTaskExtensionsTargetFrameworks>netstandard1.6;netstandard2.0;netstandard2.1</ValueTaskExtensionsTargetFrameworks>
+    <ExampleTargetFrameworks>netcoreapp2.1</ExampleTargetFrameworks>
+    <SpecsTargetFrameworks>netcoreapp1.0;netcoreapp2.1;netcoreapp3.0</SpecsTargetFrameworks>
+    <IntegrationTestsTargetFrameworks>netcoreapp1.0;netcoreapp2.1;netcoreapp3.0</IntegrationTestsTargetFrameworks>
+    <UnitTestsTargetFrameworks>netcoreapp1.0;netcoreapp2.1;netcoreapp3.0</UnitTestsTargetFrameworks>
+    <AnalyzerTestsTargetFrameworks>netcoreapp2.1;netcoreapp3.0</AnalyzerTestsTargetFrameworks>
+    <TestHelpersTargetFrameworks>netstandard1.6;netstandard2.0</TestHelpersTargetFrameworks>
+    <ApprovalTestsTargetFrameworks>netcoreapp3.0</ApprovalTestsTargetFrameworks>
+  </PropertyGroup>
+
+  <!-- .NET Framework only -->
+  <PropertyGroup Condition=" '$(TargetFrameworkSet)' == 'netfx' ">
+    <FakeItEasyTargetFrameworks>net45</FakeItEasyTargetFrameworks>

```

Comment:
and net40 as well?

Reply:
Since the goal was to make the build faster, I thought it would make sense to include as few TFMs as possible. Is there a scenario where we would want to build all netfx targets, and nothing else?
But the target sets I defined here are mostly examples anyway. The idea is that we can add new ones as needed.

Reply:
Ah. I was taking "netcoreall" for comparison, without paying attention to the "all" on the end of that.
I don't mind excluding net40 from the set, but maybe in this case, rename to net45, since we're closer to a "netcore2.1" situation than a "netcoreall" situation?

Reply:
Good point. I don't think a `netfxall` is very useful. I intended `netcoreall` to be the default for non-Windows build (could be set automatically based on the OS)